org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/MapExtensions.java

```diff
@@ -110,4 +122,484 @@ public boolean apply(Map.Entry<K, V> input) {
+	 * </p>
+	 *
+	 * <p>
+	 * Even if the key of the right operand exists in the left operand, the value in the right operand is preferred.
+	 * </p>
+	 *
+	 * @param <K> - type of the map keys.
+	 * @param <V> - type of the map values.
+	 * @param left the map to consider.
+	 * @param right the entry (key, value) to add into the map.
+	 * @return an immutable map with the content of the map and with the given entry.
+	 * @throws IllegalArgumentException - when the right operand key exists in the left operand.
+	 * @since 2.14
+	 */
+	@Pure
+	@Inline(value = "$3.union($1, $4.singletonMap($2.getKey(), $2.getValue()))",
+			imported = { MapExtensions.class, Collections.class })
+	public static <K, V> Map<K, V> operator_plus(Map<K, V> left, final Pair<? extends K, ? extends V> right) {
+		return union(left, Collections.singletonMap(right.getKey(), right.getValue()));
+	}
+
+	/**
+	 * Merge the two maps.
+	 *
+	 * <p>
+	 * The replied map is a view on the given map. It means that any change
+	 * in the original map is reflected to the result of this operation.
+	 * </p>
+	 *
+	 * <p>
+	 * If a key exists in the left and right operands, the value in the right operand is preferred.
+	 * </p>
+	 *
+	 * @param <K> - type of the map keys.
+	 * @param <V> - type of the map values.
+	 * @param left the left map.
+	 * @param right the right map.
+	 * @return a map with the merged contents from the two maps.
+	 * @throws IllegalArgumentException - when a right operand key exists in the left operand.
+	 * @since 2.14
+	 */
+	@Pure
+	@Inline(value = "$3.union($1, $2)", imported = MapExtensions.class)
+	public static <K, V> Map<K, V> operator_plus(Map<K, V> left, Map<? extends K, ? extends V> right) {
+		return union(left, right);
+	}
+
+	/**
+	 * Remove a key from the given map.
+	 *
+	 * @param <K> - type of the map keys.
+	 * @param <V> - type of the map values.
+	 * @param map the map to update.
+	 * @param key the key to remove.
+	 * @return the removed value, or <code>null</code> if the key was not
+	 *     present in the map.
+	 * @since 2.14
+	 */
+	@Inline(value = "$1.remove($2)", statementExpression = true)
+	public static <K, V> V operator_remove(Map<K, V> map, K key) {
+		return map.remove(key);
+	}
+
+	/**
+	 * Remove the given pair into the map.
+	 *
+	 * <p>
+	 * If the given key is inside the map, but is not mapped to the given value, the
+	 * map will not be changed.
+	 * </p>
+	 *
+	 * @param <K> - type of the map keys.
+	 * @param <V> - type of the map values.
+	 * @param map the map to update.
+	 * @param entry the entry (key, value) to remove from the map.
+	 * @return {@code true} if the pair was removed.
+	 * @since 2.14
+	 */
+	public static <K, V> boolean operator_remove(Map<K, V> map, Pair<? extends K, ? extends V> entry) {
+		//TODO use the JRE 1.8 API: map.remove(entry.getKey(), entry.getValue());
+		final V value = map.get(entry.getKey());
+		if (value != null && Objects.equal(value, entry.getValue())) {
```

Comment:
What if the value was null?

Reply:
I will copy/paste the code of the JRE 1.8 function that is solving the issue you have mentionned.